camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java

```diff
@@ -392,6 +399,49 @@ public int compare(FileStatus f1, FileStatus f2) {
     }
   }
 
+  private void checkIfTooManySkippedMsg(Counters counters) {
+    double maxPercentSkippedSchemaNotFound = Double.parseDouble(props.getProperty(ETL_MAX_PERCENT_SKIPPED_SCHEMANOTFOUND,
+        ETL_MAX_PERCENT_SKIPPED_SCHEMANOTFOUND_DEFAULT));
+    double maxPercentSkippedOther = Double.parseDouble(props.getProperty(ETL_MAX_PERCENT_SKIPPED_OTHER,
+        ETL_MAX_PERCENT_SKIPPED_OTHER_DEFAULT));
+
+    long actualSkippedSchemaNotFound = 0;
+    long actualSkippedOther = 0;
+    long actualDecodeSuccessful = 0; 
+    for (String groupName : counters.getGroupNames()) {
+      if (groupName.equals(EtlRecordReader.KAFKA_MSG.class.getName())) {
+        CounterGroup group = counters.getGroup(groupName);
+        for (Counter counter : group) {
+          if (counter.getDisplayName().equals(EtlRecordReader.KAFKA_MSG.DECODE_SUCCESSFUL.toString())) {
+            actualDecodeSuccessful = counter.getValue();
+          } else if (counter.getDisplayName().equals(EtlRecordReader.KAFKA_MSG.SKIPPED_SCHEMA_NOT_FOUND.toString())) {
+            actualSkippedSchemaNotFound = counter.getValue();
+          } else if (counter.getDisplayName().equals(EtlRecordReader.KAFKA_MSG.SKIPPED_OTHER.toString())) {
+            actualSkippedOther = counter.getValue();
+          }
+        }
```

Comment:
`actualSkipped > actualDecodeSuccessful` simply means more than half of the msg were skipped. It will be caught when camus job examines the percentage of skipped messages (as long as the max allowed percentage < 50%)